### ISSUE TITLE
Publish new gem with subscribe support

### ISSUE THREAD

**Issue by quark-zju** [veelenga/i3ipc-ruby#4]
Hi, thanks for the gem! It makes customizing i3 much easier. Could you publish a new version with the subscribe support? Thanks!

**Comment by veelenga** [comment]
Latest version published: `gem install i3ipc -v '0.2.0'`.

**Comment by quark-zju** [comment]
Thanks!

FYI I am trying to make i3 more scriptable. Things like 
- close all gvim windows in current workspace
- when a gvim window is opened in a hsplit container, and if the container has only terminals, move the newly created gvim right
- let the user pick a window in current workspace and focus it
- do a notify-send if a terminal title changes to some pattern
- etc.

will be doable in short ruby scripts.

Initially I'd like to use old i3 3.x column-like behavior, the upstream is [not interested](https://github.com/i3/i3/issues/2100) about the idea so I have to do it externally. The script looks like:

``` ruby
# tl;dr Bring back most of the column behavior of i3 3.x / wmii, in i3 4.x
#
# Auto run "split v" on new (or moved) windows which is the child
# of the top-level container and has "splith" layout.
#
# For example,
#
#   +---------+   +---------+   +---------+   +---------+   +----+----+ 
#   |         |   |.........|   |         |   |         |   |    |....| 
#   |         | 1 |.........| 2 |         | 3 +---------+ 4 |    |....| 
#   |         | > |.........| > +---------+ > |         | > +----+....| 
#   |         |   |.........|   |.........|   +---------+   |    |....| 
#   |         |   |.........|   |.........|   |.........|   |    |....| 
#   +---------+   +=========+   +---------+   +---------+   +----+====+ 
#
#   1. Create a new window in an empty workspace, the window will have
#      "split v" layout, regardless of "default_orientation" option.
#   2. Create a second window, no auto "split v" this time.
#   3. Create another one, works as expected.
#   4. Move the window right. It will have an auto "split v" since it is
#      the direct child of the "split h" workspace container (new column).
#
# The difference with the column behavior is that moving a bottom window in
# a column down will implicitly create a "split v" parent container.

on_window_change ['new', 'move'] do |i3, reply|
  node = reply.container
  workspace = i3.workspace_of(node)
  container = i3.inner_container_of(workspace)
  parent = i3.parent_of(node)
  if parent == container && parent.layout == 'splith'
    i3.logger.info "split v for \"#{node.name}\""
    i3.command "[con_id=#{node.id}] split v"
  end
end
```

The project is small and I'm doing some paperwork before I can make it public.

**Comment by veelenga** [comment]
Sounds good! Be free to request new features or send PRs for i3ipc-ruby if needed. Happy coding ;)
